app/scripts/controllers/transactions/lib/util.js

```diff
@@ -277,3 +277,15 @@ export function getFinalStates() {
     TRANSACTION_STATUSES.DROPPED, // the tx nonce was already used
   ];
 }
+
+/**
+ * Normalizes tx receipt gas used to be a hexadecimal string.
+ * It seems that sometimes the numerical values being returned from
+ * this.query.getTransactionReceipt are BN instances and not strings.
+ *
+ * @param {string or BN instance} gasUsed
+ * @returns normalized gas used as hexadecimal string
+ */
+export function normalizeTxReceiptGasUsed(gasUsed) {
+  return typeof gasUsed === 'string' ? gasUsed : gasUsed.toString(16);
```

Comment:
I think I mentioned this somewhere else but `string` has a `toString` method, so I don't think we need the type check.

Reply:
You mentioned it in this thread: https://github.com/MetaMask/metamask-extension/pull/12676/files/dddef0a043450bd69d1fbc00bf2d9cc98243163f#r785919603

Reply:
Let me know if the responses answer why the type check is needed. But I added this code based on @danjm comment to refactor the duplicate code into a separate function.